### DISCUSSION TITLE
whether Andes works on windows platform

### DISCUSSION THREAD

**Discussion by SanjeevKumarGoyal** [CURENT/andes#246] [General]
Good Morning,
i am looking for help to know about Andes, whether it works on windows platform or Linux platform, or both of them?
If it works on windows platform, is it possible to get DLL to include Andes and work on it?
Best Regards,
Sanjeev

**Comment by sanurielf** [comment]
AFAIK it works in multiple platforms including Windows, Linux, OSx.

**Reply by SanjeevKumarGoyal** [reply]
Thank you so much for your response.
Is their any DLL which i need to register to start working with it.
Regards,
Sanjeev

**Comment by cuihantao** [comment]
All features of ANDES work on Windows thanks to the KVXOPT package 😄 @sanurielf 

ANDES is being used in a project primarily based on Windows.

**Reply by SanjeevKumarGoyal** [reply]
Thank you so much for confirming about windows.
Regards,
Sanjeev

**Comment by cuihantao** [comment]
As a Python package, it does not compile to a single DLL (without using a compiler like Nuitka, which I haven’t had any luck of success). 

If you are using it from Python, you can simply import it. For other languages, you will need a middleware that bridges the host language with Python.

**Comment by cuihantao** [comment]
@SanjeevKumarGoyal A Python package does not compile to DLLs. In which language do you intend to use ANDES?

**Reply by SanjeevKumarGoyal** [reply]
I am working on visual studio 2017 and using c# (.net)

**Reply by cuihantao** [reply]
Got it. You might want to look up general articles on how to call Python in C#.

**Reply by SanjeevKumarGoyal** [reply]
Please share the reference articles on this, those would be helpful at initial stage. is that possible to move to ANDES to use later in c#.

**Reply by cuihantao** [reply]
You can search for that on Google. I have zero experience in C# so won't be able to do much.

**Reply by SanjeevKumarGoyal** [reply]
Thank you so much!
let me summarize another item, Please help me on this; 
I have developed application in visual studio using c# (.net).
I want to work on ANDES which supports windows platform as well.
what are the primary requirements to do this?
Thank you so much!

**Reply by cuihantao** [reply]
Looks to me it’s the same issue as your original one. I still suggest you look up general tutorials on installing Python packages and using Python in C#.

**Reply by SanjeevKumarGoyal** [reply]
Thank you so much.
Okay i will start exploring them and let you know if need more help on this.
